### PR TITLE
create port enhancement

### PR DESCRIPTION
**Pull request by ring630**
```
        U7 = edbapp.components["U7"]
        U7.pins["G7"].create_port()
        port = U7.pins["F7"].create_port(reference=U7.pins["E7"])
        port.is_circuit_port = True
        _, pin_group = edbapp.siwave.create_pin_group_on_net(
            reference_designator="U7", net_name="GND", group_name="U7_GND"
                                              )
        U7.pins["F7"].create_port(reference=pin_group)
```

### CODE REVIEW

**Comment by ring630** [comment]
@SMoraisAnsys this is edb related.

File: pyaedt/edb.py [hunk 1]
```diff
@@ -3639,3 +3643,36 @@ def _get_connected_ports_from_multizone_cutout(self, terminal_info_dict):
...
                                                 connected_ports_list.append((port1_connexion, port2_connexion))
             return connected_ports_list
+
+    @pyaedt_function_handler
+    def create_port(self, terminal, ref_terminal=None, is_circuit_port=False):
+        """Create a port between two terminals.
+
+        Parameters
+        ----------
+        terminal : class:`pyaedt.edb_core.edb_data.terminals.EdgeTerminal`,
```
Comment (PipKat) [inline pyaedt/edb.py:3653]: Please add brief descriptions for the two terminal parameters.

File: pyaedt/edb_core/edb_data/padstacks_data.py [hunk 2]
```diff
@@ -1018,12 +1018,39 @@ def _create_terminal(self, name=None):
...
+        return port
+
+    @pyaedt_function_handler
+    def create_port(self, name=None, reference=None, is_circuit_port=False):
+        """Create a port on the padstack.
+
+        Parameters
+        ----------
+        name : str, optional
+            Name of the port.
```
Comment (PipKat) [inline pyaedt/edb_core/edb_data/padstacks_data.py:1033]:
```suggestion
            Name of the port. The default is ``None``, in which case a name is automatically assigned.
```
The description of the behavior that occurs when ``None`` is used is a guess. Please revise if it is wrong.

File: pyaedt/edb_core/edb_data/padstacks_data.py [hunk 3]
```diff
@@ -1018,12 +1018,39 @@ def _create_terminal(self, name=None):
...
+        reference : class:`pyaedt.edb_core.edb_data.nets_data.EDBNetsData`,
+                    class:`pyaedt.edb_core.edb_data.padstacks_data.EDBPadstackInstance`,
+                    class:`pyaedt.edb_core.edb_data.sources.PinGroup`,
```
Comment (PipKat) [inline pyaedt/edb_core/edb_data/padstacks_data.py:1036]: Add a brief description for "reference" parameter.

Comment (PipKat) [inline pyaedt/edb_core/edb_data/padstacks_data.py:1036]:
```suggestion
                    class:`pyaedt.edb_core.edb_data.sources.PinGroup`, optional
```

File: pyaedt/edb_core/edb_data/padstacks_data.py [hunk 4]
```diff
@@ -1164,6 +1191,7 @@ def in_polygon(self, polygon_data, include_partial=True, simple_check=False):
     @property
     def pin(self):
         """Return Edb padstack object."""
```
Comment (PipKat) [inline pyaedt/edb_core/edb_data/padstacks_data.py:1193]:
```suggestion
        """EDB padstack object."""
```

File: pyaedt/edb_core/edb_data/sources.py [hunk 5]
```diff
@@ -274,34 +275,52 @@ def net(self, value):
     def net_name(self):
         return self._edb_pin_group.GetNet().GetName()
 
+    @property
+    def terminal(self):
+        """Get terminal."""
```
Comment (PipKat) [inline pyaedt/edb_core/edb_data/sources.py:280]:
```suggestion
        """Terminal."""
```
Properties are described as nouns.

File: pyaedt/edb_core/edb_data/sources.py [hunk 6]
```diff
@@ -274,34 +275,52 @@ def net(self, value):
...
-                is_reference,
-            )
-        else:
+    def _create_terminal(self, name=None):
+        """Create a terminal on the pin group.
+
+        Parameters
+        ----------
+        name : str, optional
+            Name of the terminal.
```
Comment (PipKat) [inline pyaedt/edb_core/edb_data/sources.py:293]:
```suggestion
            Name of the terminal. The default is ``None``, in which case a name is
            automatically assigned.
```

File: pyaedt/edb_core/edb_data/terminals.py [hunk 7]
```diff
@@ -562,3 +562,32 @@ class PinGroupTerminal(Terminal):
...
+        Parameters
+        ----------
+        name : str
+            Name of the terminal.
+        net_name : str
+            Name of the net.
+        pin_group_name : str,
+            Name of the pin group.
+        is_ref : bool, optional
+            Whether it is a reference terminal.
```
Comment (PipKat) [inline pyaedt/edb_core/edb_data/terminals.py:579]:
```suggestion
            Whether it is a reference terminal. The default is ``False``.
```

File: pyaedt/edb_core/edb_data/sources.py [hunk 8]
```diff
@@ -274,34 +275,53 @@ def net(self, value):
...
+            name = self.name
+
+        if pg_term:
             return pg_term
+        else:
+            from pyaedt.edb_core.edb_data.terminals import PinGroupTerminal
+
+            term = PinGroupTerminal(self._pedb)
+
+            term = term.create(name, self.net_name, self.name)
```
Comment (svandenb-dev) [inline pyaedt/edb_core/edb_data/sources.py:311]: I don't quite understand. the method _create_terminal creates actually a PingroupTerminal whichius different. I found it confusing

File: pyaedt/edb_core/edb_data/sources.py [hunk 9]
```diff
@@ -274,34 +275,53 @@ def net(self, value):
     def net_name(self):
         return self._edb_pin_group.GetNet().GetName()
 
+    @property
+    def terminal(self):
+        """Terminal."""
+        from pyaedt.edb_core.edb_data.terminals import PinGroupTerminal
+
+        term = PinGroupTerminal(self._pedb, self._edb_pin_group.GetPinGroupTerminal())
```
Comment (svandenb-dev) [inline pyaedt/edb_core/edb_data/sources.py:283]: Here property terminal is returning a PingroupTerminal ?

**Review by svandenb-dev — CHANGES_REQUESTED** [review]

File: pyaedt/edb_core/edb_data/sources.py (hunk 8 above)
Comment (svandenb-dev) [reply pyaedt/edb_core/edb_data/sources.py:311]: ok with this one LGTM

**Review by svandenb-dev — APPROVED** [review]

**Pull request merged**